libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -702,15 +778,30 @@ string ABIFunctions::abiEncodingFunctionSimpleArray(
 		templ("return", dynamic ? " -> end " : "");
 		templ("assignEnd", dynamic ? "end := pos" : "");
 		templ("lengthFun", arrayLengthFunction(_from));
-		if (_to.isDynamicallySized())
+		if (_to.isDynamicallySized() && !_options.dynamicInplace)
 			templ("storeLength", "mstore(pos, length) pos := add(pos, 0x20)");
 		else
 			templ("storeLength", "");
 		templ("dataAreaFun", arrayDataAreaFunction(_from));
-		templ("elementEncodedSize", toCompactHexWithPrefix(_to.baseType()->calldataEncodedSize()));
+		if (!usesTail)
+		{
+			if (_options.dynamicInplace && dynamicBase)
+			{
+				templ("assignToPos", "pos :=");
+				templ("incrementPos", "");
+			}
+			else
```

Comment:
Ok, so this is the case of ``!usesTail && !(_options.dynamicInplace && dynamicBase)``..
So... ``!(dynamicBase && !_options.dynamicInplace) && !(_options.dynamicInplace && dynamicBase)``...
So... ``(!dynamicBase || _options.dynamicInplace) && (!_options.dynamicInplace || !dynamicBase)``
So... ``!dynamicBase``, right? Maybe that's worth a comment..

Reply:
Maybe it's easier to start with that case as ``if (!dynamicBase)`` and have an ``else if(_options.dynamicInPlace)`` which should be the other case above? Not sure, but the logic is a bit hard to read as it is.